include/picolibrary/microchip/megaavr/peripheral/spi.h

```diff
@@ -0,0 +1,487 @@
+         */
+        SPCR( SPCR && ) = delete;
+
+        /**
+         * \todo #27
+         */
+        SPCR( SPCR const & ) = delete;
+
+        ~SPCR() = delete;
+
+        /**
+         * \todo #27
+         *
+         * \return
+         */
+        auto operator=( SPCR && ) = delete;
+
+        /**
+         * \todo #27
+         *
+         * \return
+         */
+        auto operator=( SPCR const & ) = delete;
+
+        using Register<std::uint8_t>::operator=;
+
+        /**
+         * \brief Configure the SPI for use as an SPI controller.
+         */
+        void configure_controller() noexcept
+        {
+            *this = Mask::MSTR;
+        }
+
+        /**
+         * \brief Configure the SPI for use as an SPI device.
+         */
+        void configure_device() noexcept
+        {
+            *this = 0;
+        }
+
+        /**
+         * \brief Enable the SPI.
+         */
+        void enable() noexcept
+        {
+            *this |= Mask::SPE;
+        }
+
+        /**
+         * \brief Disable the SPI.
+         */
+        void disable() noexcept
+        {
+            *this &= ~Mask::SPE;
+        }
+
+        /**
+         * \brief Enable the SPI interrupt.
+         */
+        void enable_interrupt() noexcept
+        {
+            *this |= Mask::SPIE;
+        }
+
+        /**
+         * \brief Disable the SPI interrupt.
+         */
+        void disable_interrupt() noexcept
+        {
+            *this &= static_cast<std::uint8_t>( ~Mask::SPIE );
+        }
+
+        /**
+         * \brief Configure SPI data exchange.
+         *
+         * \param[in] clock_rate The desired clock rate.
+         * \param[in] clock_polarity The desired clock polarity.
+         * \param[in] clock_phase The desired clock phase.
+         * \param[in] bit_order The desired data exchange bit order.
+         */
+        void configure( Clock_Rate clock_rate, Clock_Polarity clock_polarity, Clock_Phase clock_phase, Bit_Order bit_order ) noexcept;
+    };
+
+    /**
+     * \brief SPI Status Register (SPSR).
+     *
+     * This register has the following fields:
+     * - Double SPI Speed (SPI2X)
+     * - Write Collision (WCOL)
+     * - SPI Interrupt (SPIF)
+     */
+    class SPSR : public Register<std::uint8_t> {
+      public:
+        /**
+         * \brief Field sizes.
+         */
+        struct Size {
+            static constexpr auto SPI2X    = std::uint_fast8_t{ 1 }; ///< SPI2X.
+            static constexpr auto RESERVED = std::uint_fast8_t{ 5 }; ///< Reserved.
+            static constexpr auto WCOL     = std::uint_fast8_t{ 1 }; ///< WCOL.
+            static constexpr auto SPIF     = std::uint_fast8_t{ 1 }; ///< SPIF.
+        };
+
+        /**
+         * \brief Field bit positions.
+         */
+        struct Bit {
+            static constexpr auto SPI2X = std::uint_fast8_t{}; ///< SPI2X.
+            static constexpr auto RESERVED = std::uint_fast8_t{ SPI2X + Size::SPI2X }; ///< RESERVED.
```

Comment:
`RESERVED` should be lower case in the comment for consistency with the other comments.